File: Backend/barbershop/barbershop/Models/ResponseDTOs/ProductDTO.cs
using barbershop.Models.Entitys;

namespace barbershop.Models.ResponseDTOs
{
    public class ProductDTO
    {
        public int ProductId { get; set; }
        public string? ProductName { get; set; }
        public int? ProductTypeId { get; set; }
        public string? Instruction { get; set; }
        public bool? IsActive { get; set; }
        public int? SizeId { get; set; }
        public string? ProductImg { get; set; }
        public int? BrandId { get; set; }
        public bool? IsOutstanding { get; set; }
        public Brand? Brand { get; set; }
        public ProductPriceDTO? ProductPriceDTO { get; set; }
        public ProductType? ProductType { get; set; }
        public List<SizeDTO>? SizeDTOs { get; set; }
    }
}

File: Frontend/barbershop/src/types/ResponseDTOs/productDTO.ts
import type { BrandDTO } from './brandDTO';
import type { InventoryDTO } from './inventoryDTO';
import type { PaymentServiceDTO } from './paymentServiceDTO';
import type { ProductPriceDTO } from './productPriceDTO';
import type { ProductTypeDTO } from './productTypeDTO';
import type { SizeDTO } from './sizeDTO';

export type ProductDTO = {
    productId: number;
    productName?: string;
    productTypeId?: number;
    instruction?: string;
    isActive?: boolean;
    sizeId?: number;
    productImg?: string;
    brandId?: number;
    isOutstanding?: boolean;
    brandDTO?: BrandDTO;
    inventoryDTO?: InventoryDTO[];
    paymentServiceDTO?: PaymentServiceDTO[];
    productPriceDTO?: ProductPriceDTO;
    productTypeDTO?: ProductTypeDTO;
    sizeDTO?: SizeDTO[];

    //Thêm
    sizeName?: string;
    quantity?: number;
};
